### PR TITLE
run route-sync tests as part of CI test script

### PR DESCRIPTION
**Pull request by johnsonj**
***trying out PR on a PR to see how that feels***

This script is expected to run under the `pcfkubo/kubo-ci` docker image by CI which
contains the appropriate version of go and ginkgo. It can also be used
by local developers for inner loop testing.

The following is validated by this change:
 - ginkgo -r .: correctness of the tests
 - ginkgo -r -race .: no observable race conditions
 - go build: entire package compiles to exercise not tested code (eg
   main.go). We want to add test coverage to our main code but there
   will always be an entry point that can only be validated by
   compilation.

[#137357741]

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@johnsonj Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=pivotal-cf-experimental/kubo-release&pullRequestId=28)!

[Click here](https://cla.pivotal.io/sync/pivotal?repositoryId=pivotal-cf-experimental/kubo-release&pullRequestId=28) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@johnsonj Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=pivotal-cf-experimental/kubo-release&pullRequestId=28)!

File: script/run_tests [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
\ No newline at end of file
+test_route_sync() {
+  pushd "${project_root}/src/route-sync" > /dev/null
+    export GOPATH=${project_root}
+    echo "Running ginkgo"
+    ginkgo -r .
+    echo "Running ginkgo to check race conditions"
+    ginkgo -r -race .
+    echo "Compiling final binary"
+    go build
```
Comment (jhvhs) [inline script/run_tests]: Why do we have to build here?

**Comment by johnsonj** [comment]
Please see the pr description for my rationale there

File: script/run_tests (hunk 1 above)
Comment (johnsonj) [inline script/run_tests]: please see the PR description for rationale

**Comment by johnsonj** [comment]
ping @jhvhs

**Comment by alex-slynko** [comment]
I left the comment above. In the end, `go build` tests compilability of main.go.
I hope we will be able to cover it with some linter tests.

**Pull request merged**